### PR TITLE
Adopt aiohttp>=3.0.0 (#519)

### PR DESCRIPTION
**Pull request by Gr1N**
Dropped support of Python 3.4 and aiohttp < 3.0.0.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -9,16 +9,16 @@
...
     'multidict>=2.1.4',
     'wrapt>=1.10.10',
     'packaging>=16.8',
 ]
 
 PY_VER = sys.version_info
 
-if not PY_VER >= (3, 4, 1):
-    raise RuntimeError("aiobotocore doesn't support Python earlier than 3.4")
+if not PY_VER >= (3, 5, 0):
```
Comment (asvetlov) [inline setup.py]: 3.5.3 actually. The same in all other places

Comment (Gr1N) [reply setup.py]: fixed, thanks

File: tests/test_patches.py [hunk 2]
```diff
@@ -28,19 +28,19 @@
...
 
     # for our replacement of read()
-    DataQueue: {'8ad4d5df1d016547daea6389707bc656630582e5'},
+    DataQueue: {'be516f7bcdbf5235218261d8ed1f490d299f611d'},
 
     # for our patch of _wait
-    StreamReader: {'c0a9a31a8c3e550de5985ab642028983f709b37b'},
+    StreamReader: {'d4ffb6ae823ef4bfd810aade8601ba7b01aa08ec'},
 
     # for digging into _protocol ( 2.1.x, 2.2.x )
```
Comment (thehesiod) [inline tests/test_patches.py]: can remove 2.1.x, 2.2.x

**Comment by thehesiod** [comment]
cool looks good!  I validated the changes to aiobotocore and looks like there's no impact to us, yay.  just have one comment to remove a comment :)  Let's see what @jettify says too.

**Comment by thehesiod** [comment]
would be good to bump to the latest botocore too, but we can do that in another PR

File: tests/test_patches.py (hunk 2 above)
Comment (Gr1N) [reply tests/test_patches.py]: removed

File: tests/mock_server.py [hunk 3]
```diff
@@ -45,8 +46,12 @@ def _run(self):
         app.router.add_route('*', '/{anything:.*}', self.stream_handler)
 
         try:
-            aiohttp.web.run_app(app, host=host, port=self._port,
-                                loop=self._loop, handle_signals=False)
+            # We need to mock `.get_event_loop` function and return
+            # `self._loop` explicitly because from `aiohttp>=3.0.0` we can't
+            # pass `loop` as a kwargs into `run_app`.
+            with mock.patch('asyncio.get_event_loop', return_value=self._loop):
+                aiohttp.web.run_app(app, host=host, port=self._port,
```
Comment (jettify) [inline tests/mock_server.py]: We can use new AppRunner stuff as result we do not need ugly patch
https://docs.aiohttp.org/en/stable/web_advanced.html?highlight=AppRunner#application-runners

**Comment by jettify** [comment]
Changes looks good to me as well, one nitpick. We can merge PR as it is and handle runner stuff in separate PR

**Comment by Gr1N** [comment]
@jettify maybe it's bad, but I don't want to spend time on `AppRunner`. If it possible please merge this PR and migrate to `AppRunner` in separate PR.

**Comment by jettify** [comment]
thanks!

**Pull request merged**